ui/apps/platform/src/Containers/Collections/CollectionForm.tsx

```diff
@@ -152,255 +125,172 @@ function CollectionForm({
-                <DrawerContent
-                    panelContent={
-                        <DrawerPanelContent
-                            style={{
-                                borderLeft: 'var(--pf-global--BorderColor--100) 1px solid',
-                            }}
+        <Formik
+            initialValues={
+                action.type === 'clone'
+                    ? { ...initialData, name: `${initialData.name} (COPY)` }
+                    : initialData
+            }
+            onSubmit={(collection, { setSubmitting }) => {
+                onSubmit(collection).catch(() => {
+                    setSubmitting(false);
+                });
+            }}
+            validationSchema={yup.object({
+                name: yup.string().trim().required(),
+                description: yup.string(),
+                embeddedCollectionIds: yup.array(yup.string().trim().required()),
+                resourceSelector: yup.object().shape({
+                    Deployment: yupResourceSelectorObject(),
+                    Namespace: yupResourceSelectorObject(),
+                    Cluster: yupResourceSelectorObject(),
+                }),
+            })}
+        >
+            {({
+                values,
+                isValid,
+                errors,
+                handleChange,
+                handleBlur,
+                setFieldValue,
+                submitForm,
+                isSubmitting,
+            }) => (
+                <Form className="pf-u-background-color-200">
+                    <Flex
+                        className="pf-u-p-lg"
+                        spaceItems={{ default: 'spaceItemsMd' }}
+                        direction={{ default: 'column' }}
+                    >
+                        <Flex
+                            className="pf-u-background-color-100 pf-u-p-lg"
+                            direction={{ default: 'column' }}
+                            spaceItems={{ default: 'spaceItemsMd' }}
                         >
-                            <DrawerHead>
-                                <Title headingLevel="h2">Collection results</Title>
-                                <Text>See a live preview of current matches.</Text>
-                                <DrawerActions>
-                                    <DrawerCloseButton onClick={() => toggleDrawer(false)} />
-                                </DrawerActions>
-                            </DrawerHead>
-                            <DrawerPanelBody className="pf-u-h-100" style={{ overflow: 'auto' }}>
-                                <CollectionResults />
-                            </DrawerPanelBody>
-                        </DrawerPanelContent>
-                    }
-                >
-                    <DrawerContentBody className="pf-u-background-color-100 pf-u-display-flex pf-u-flex-direction-column">
-                        {headerContent}
-                        {initialData instanceof AggregateError ? (
```

Comment:
Ternary expression caused change from `useFormik` hook to `Formik` element and render prop pattern.

After this refactoring step, what are pro and con for return of the hook. It sounds like a movie ;)

Reply:
Haha, so this was a change that I've been thinking of for a while and decided to make since this code was already in flux.

Two main reasons:
- The [documentation on the Formik site](https://formik.org/docs/api/useFormik#use-cases-for-useformik) is pretty insistent that this is not the correct use case for this hook.
- I was testing usage of some of the other Formik components (`<Field>`, `<FieldArray>`), but none of them work since the Formik context is not available when using this hook. Additionally I think it might be cleaner to access some of the form values/errors in child components via the `useFormikContext` hook.

Reply:
Thank you for explaining.

* Maintainer wrote that when render prop pattern was at its height, but React Router is example of comparable package which moved to hooks without looking back.
* Yeah, Linda needed context for policies to compose container form elements. A word of caution is to render Formik-specific elements only if package-specific abstraction adds enough value to balance need for team to understand it versus Formik state plus PatternFly form elements which we must all know well.